### PR TITLE
Add skipLogCollection input for E2E tests

### LINKED ISSUE #2686: Add skipLogCollection flag to E2E tests
/kind feature

**Describe the solution you'd like**
In order to add more flexibility to running E2E tests locally, we should add a skipLogCollection flag to the E2E test options. This is also a temporary solution to tests taking an extremely long time to run on local dev environments with Azure subscriptions that block SSH.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds a `skipLogCollection` field for the E2E test cleanup input. This is helpful for people with internal dev subscriptions to be able to run specific tests locally without waiting for the log collection to time out.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2686

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -655,6 +656,7 @@ test-e2e-run: generate-e2e-templates install-tools ## Run e2e tests.
     $(GINKGO) -v -trace -tags=e2e -focus="$(GINKGO_FOCUS)" -skip="$(GINKGO_SKIP)" -nodes=$(GINKGO_NODES) --noColor=$(GINKGO_NOCOLOR) $(GINKGO_ARGS) ./test/e2e -- \
     	-e2e.artifacts-folder="$(ARTIFACTS)" \
     	-e2e.config="$(E2E_CONF_FILE_ENVSUBST)" \
+		-e2e.skip-log-collection=$(SKIP_LOG_COLLECTION) \
```
Comment (mboersma) [inline Makefile:659]: Some extra whitespace here.

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

**Review by CecileRobertMichon — COMMENTED** [review]
/retest

**Pull request merged**